python/py-buildbot-pkg/Portfile

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-buildbot-pkg
+version             2.3.1
+revision            0
+
+maintainers         nomaintainer
+description         Buildbot packaging tools
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            http://buildbot.net/
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            ${python.rootname}-${version}
+
+license             GPL-2
+
+checksums           sha256  25968ace0c62cb773ed85d4ddbe07fd5aee68f4455909243ffb3ac12608cf82e \
+                    rmd160  b917b5d58d24874068eede4bc7312cb3c58185e1 \
+                    size    4754
+
+python.versions     36 37
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:py${python.version}-setuptools
```

Comment:
This seems unusual. Normally setuptools is required to run `setup.py`, but not afterwards. Can you check this again?

Reply:
I would say this is now a question for @Korusuke / @reneeotten / @Steap. This is in fact literally what `upt` returns, and I would claim we need to fix the tool (the source of the "issue"). It could have something to do with pypi not providing sufficient information. I remember this discussion from another PR, but I probably won't be able to find it in the first shot.

In any case it is unusual that the port would a dependency inside **both** `build` and `lib`, as that's overlapping.

Reply:
Yeah, this is a known issue and you are right we did have a discussion on this.
The issue is that pypi doesn't give `setuptools` as a dependency if it is build dependency but if it's a lib dependency then it does return `setuptools`. 
So for such special cases, changes will be needed to be handled by upt itself and I will add this soon.

Was not sure about this until now, also this should be a rare issue as I tried random 50 ports from MacPorts of pypi and only 2 of them were such outliers.

Reply:
Checked again, the runtime dependency is expected. I changed the code, so that it's not overlapping. Now it's once declared as a build dependency and once as a runtime dependency (rather than lib).